packages/server/tests/MpeRoomCreateInvitation.test.ts

```diff
@@ -0,0 +1,288 @@
+            );
+            assert.isTrue(creatorReceivedMpeRoomInvitationSpy.notCalled);
+        });
+
+        const allInvitations = await MpeRoomInvitation.all();
+        assert.equal(allInvitations.length, 1);
+
+        const mpeRoom = await MpeRoom.findOrFail(roomID);
+        await mpeRoom.load('invitations');
+
+        assert.equal(mpeRoom.invitations.length, 1);
+    });
+
+    test('It should fail to invite user as invitedUser is creator himself', async (assert) => {
+        const creatorUserID = datatype.uuid();
+        const roomID = datatype.uuid();
+
+        const socket = await createUserAndGetSocket({
+            userID: creatorUserID,
+            mpeRoomIDToAssociate: [
+                {
+                    roomID,
+                },
+            ],
+        });
+
+        const mpeRoomReceivedInvitationSpy = createSpyOnClientSocketEvent(
+            socket,
+            'MPE_RECEIVED_ROOM_INVITATION',
+        );
+
+        try {
+            await MpeRoomsWsController.onCreatorInviteUser({
+                invitingUserID: creatorUserID,
+                invitedUserID: creatorUserID,
+                roomID,
+            });
+            assert.isTrue(false);
+        } catch ({ message }) {
+            assert.equal((await MpeRoomInvitation.all()).length, 0);
+            assert.equal(
+                message,
+                'Creator cannot invite himself in his mpe room',
+            );
+            assert.isTrue(mpeRoomReceivedInvitationSpy.notCalled);
+        }
+    });
+
+    test('It should fail to invite user as the inviting user is not the mpe room creator', async (assert) => {
+        const invitedUserID = datatype.uuid();
+        const normalUserID = datatype.uuid();
+        const creatorUserID = datatype.uuid();
+        const roomID = datatype.uuid();
+        await createUserAndGetSocket({
+            userID: creatorUserID,
+            mpeRoomIDToAssociate: [{ roomID }],
+        });
+
+        await createUserAndGetSocket({
+            userID: normalUserID,
+            mpeRoomIDToAssociate: [{ roomID }],
+        });
+
+        const invitedUserSocket = await createUserAndGetSocket({
+            userID: invitedUserID,
+        });
+
+        const invitedUserReceivedMpeRoomInvitationSpy =
+            createSpyOnClientSocketEvent(
+                invitedUserSocket,
+                'MPE_RECEIVED_ROOM_INVITATION',
+            );
+
+        try {
+            await MpeRoomsWsController.onCreatorInviteUser({
+                invitingUserID: normalUserID,
+                invitedUserID: invitedUserID,
+                roomID,
+            });
+            assert.isTrue(false);
+        } catch ({ message }) {
+            assert.equal((await MpeRoomInvitation.all()).length, 0);
+            assert.equal(
+                message,
+                'Emitter user does not appear to be the mpe room creator',
+            );
```

Comment:
Japa has a `instanceOf` assertion that we could use to check the thrown error is an instance of a custom `Error` class we would have written.
As we did in Go, it would allow assertions to do not break if the message of the error changes, because this is not necessarily what we want to test there.
Creating custom errors is quite boring and I am not saying we should refactor this code. I just wanted to point it out.

Reply:
You're completely right